pkg/skaffold/runner/context/context.go

```diff
@@ -32,6 +32,8 @@ type RunContext struct {
 	Opts *config.SkaffoldOptions
 	Cfg  *latest.Pipeline
 
+	Trigger chan bool
```

Comment:
1. I'm worried about putting channels into a shared context. Theoretically this design enables anybody can read it / mutate it, and then it gets messed up, it violates encapsulation. 
2.  Are you planning to introduce a channel for each trigger (build/deploy/sync)? Then we should rename this to `BuildTriggerChan` or similar

Reply:
the same can be said about the other structs in the run context, in theory anybody can read/mutate them, and then the skaffold pipeline gets messed up.

I wanted to make this channel read only, but it needs to be writable so we can pass the same channel to the server handler that writes to it. I think ultimately if we're careful with what we do with the runcontext throughout the code, we should be ok here.

I'll be renaming this channel in a follow up PR when I add one for a deploy trigger.

Reply:
> the same can be said about the other structs in the run context, in theory anybody can read/mutate them, and then the skaffold pipeline gets messed up.

There is a difference though: the structs can (and should) be made read-only via exposing getters  and making them private, just like the golang context object works. The getters can return a copy of the structs and then the whole structure will be immutable. Channels however - even readonly ones - won't be immutable, and if somebody else reads from it, this will be a problem. 

I feel strongly about not having the channel here. What is the benefit of creating it in the runner in the first place?

Reply:
For the record: we discussed and came to the conclusion that this is the best we can come up with currently as: 
 - my argument about structs vs channels does not stand that well: readonly channels can be made immutable as well by creating a Getter that kicks off a go routine copying from the "private" channel 
 - currently the RunContext and the Runner is our dependency injection mechanism for our components. Any other cross reference between the components (this case the Watcher and the API Server) would need some other form of injection. Currently none of these alternative avenues were attractive. 
Resolving this and LGTM! :)